esp/services/ws_workunits/ws_workunitsHelpers.hpp

```diff
@@ -198,37 +191,144 @@ struct WUComponentLogOptions
         }
         else
         {
-            if (logFilterReq.getRelativeTimeRangeBuffer() > 0 )
-                wuLogSearchTimeBuffSecs = logFilterReq.getRelativeTimeRangeBuffer();
+            if (relativeTimeBufferSecs > 0 )
+                wuLogSearchTimeBuffSecs = relativeTimeBufferSecs;
         }
+    }
 
-        if (!isEmptyString(logFilterReq.getSelectColumnModeAsString()))
+    void populateLogFilter(const char * wuid, CHttpRequest* zapHttpRequest)
+    {
+        ILogAccessFilter * logFetchFilter = getJobIDLogAccessFilter(wuid);
+
+        StringBuffer requestedLogDataFormat;
+        zapHttpRequest->getParameter("LogFilter_Format", requestedLogDataFormat);
+        if (!requestedLogDataFormat.isEmpty())
+            logDataFormat = logAccessFormatFromName(requestedLogDataFormat.str());
+
+        StringBuffer start; // Absolute query time range start in YYYY-DD-MMTHH:MM:SS
+        zapHttpRequest->getParameter("LogFilter_AbsoluteTimeRange_Start", start);
+        StringBuffer end; // Absolute query time range end in YYYY-DD-MMTHH:MM:SS
+        zapHttpRequest->getParameter("LogFilter_AbsoluteTimeRange_End", end);
+        // Query time range based on WU Time +- Buffer in seconds
+        unsigned bufferSecs = (unsigned)zapHttpRequest->getParameterInt("LogFilter_RelativeTimeRangeBuffer", 0);
+
+        populateTimeRange(start, end, bufferSecs);
+
+        //int  0 ==MIN, 1==DEFAULT, 2==ALL, 3==CUSTOM
+        int colMode = zapHttpRequest->getParameterInt("LogFilter_ColumnMode", -1);
+        if (colMode != -1)
         {
-            switch (logFilterReq.getSelectColumnMode())
-            {
-            case 0:
-                logFetchOptions.setReturnColsMode(RETURNCOLS_MODE_min);
-                break;
-            case 1:
-                logFetchOptions.setReturnColsMode(RETURNCOLS_MODE_default);
-                break;
-            case 2:
-                logFetchOptions.setReturnColsMode(RETURNCOLS_MODE_all);
-                break;
-            case 3:
-                logFetchOptions.setReturnColsMode(RETURNCOLS_MODE_custom);
-                if (logFilterReq.getCustomColumns().length() > 0 )
-                    logFetchOptions.copyLogFieldNames(logFilterReq.getCustomColumns());
-                else
-                    throw makeStringException(ECLWATCH_LOGACCESS_UNAVAILABLE, "WsWuInfo: LogFilter empty custom colums detected!");
-                break;
-            default:
-                logFetchOptions.setReturnColsMode(RETURNCOLS_MODE_default);
-                break;
-            }
+            //comma delimited list of available columns, only if ColumnMode==3
```

Comment:
I see the confusion perhaps that comment should be moved to "setReturnColumMode()"...

so we only call "setReturnColumnMode if colMod != -1, meaning the request contains a value for LogFilter_ColumnMode, that function requires a StringArray with custom columns in case colMod was set to 3...